File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.java
class Solution {
    public int characterReplacement(String s, int k) {
        Map<Character, Integer> count = new HashMap<>();
        int left = 0, maxF = 0, result = 0;

        for (int right = 0; right < s.length(); right++) {
            count.put(s.charAt(right), count.getOrDefault(s.charAt(right), 0) + 1);
            maxF = Math.max(maxF, count.get(s.charAt(right)));

            while ((right - left + 1) - maxF > k) {
                count.put(s.charAt(left), count.get(s.charAt(left)) - 1);
                left++;
            }

            result = Math.max(result, right - left + 1);
        }

        return result;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-4.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int right = 0;
        for(int pile : piles) {
            right = max(right, pile);
        }

        int left = 1, k = right;
        while(left <= right) {
            int mid = left + (right - left) / 2;

            int totalTime = 0;
            for(int pile : piles) {
                if(pile % mid != 0) {
                    totalTime += pile / mid + 1;
                } else {
                    totalTime += pile / mid;
                }
            }

            if(totalTime <= h) {
                k = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return k;
    }
};
